### PR TITLE
[GPU][TRANSFORMATIONS] Support RMS Normalization Fusion without Learnable Affine Parameters

### PR DESCRIPTION
**Pull request by andrew-k-park**
### Details:
 - This PR enhances RMS normalization fusion to support pattern without learnable affine parameter (gamma), enabling optimization of transformer architecture like LTX-Video
 - The existing RMS fusion pass only supported pattern with constant gamma parameter. However, some transformer model (e.g., LTX-Video's attention layers) use RMS normalization followed by dynamic scaling operation where the scale factor is non-constant. These pattern was previously unfused, missing optimization opportunity
 - When `elementwise_affine=False` (equivalent to [Pytorch RMS's attribute](https://docs.pytorch.org/docs/stable/generated/torch.nn.modules.normalization.RMSNorm.html)), RMS normalization does not include learnable gamma parameters. The gamma is implicitly fixed to ones, reducing the decomposed graph pattern
from:
`x → Power(2) → ReduceMean → Add(eps) → Sqrt → Divide(1/√) → Multiply(x, 1/√) → Multiply(gamma) `
to:
`x → Power(2) → ReduceMean → Add(eps) → Sqrt → Divide(1/√) → Multiply(x, 1/√)  [NO gamma multiplication] `
image-2026-01-26-22-53-05-973

### Tickets:
 - [CVS-179953](https://jira.devtools.intel.com/browse/CVS-179953)

### CODE REVIEW

File: src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp [hunk 1]
```diff
@@ -114,9 +124,24 @@ RMSFusion::RMSFusion(bool force_tail_convert, bool enable_div_x) {
...
+            }
+        } else {
+            auto input_shape = x_output.get_partial_shape();
+            if (input_shape.rank().is_dynamic() || input_shape[input_shape.size() - 1].is_dynamic()) {
+                return false;
+            }
+            auto last_dim = input_shape[input_shape.size() - 1].get_length();
+            auto gamma_shape = ov::Shape{static_cast<size_t>(last_dim)};
+            auto output_type = mul_or_div_node->get_output_element_type(0);
+            gamma_node = v0::Constant::create(output_type, gamma_shape, {1.0f});
```
Comment (e-ddykim) [inline src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp]: What do you think about adding a new constructor in the `ov::op::internal::RMS` that does not require `gamma`? Then, we don't need to create a dummy `constant` node.

Comment (andrew-k-park) [reply src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp]: applied as your suggestion

File: src/common/transformations/include/ov_ops/rms.hpp [hunk 2]
```diff
@@ -24,10 +24,23 @@ class TRANSFORMATIONS_API RMS : public ov::op::Op {
     /// \param gamma Gamma values for weight
     /// \param eps Epsilon for not dividing by zero while normalizing the value
     /// \param output_type Output element type
+    /// \param elementwise_affine A boolean value that when set to True, RMS has learnable affine parameters
     RMS(const Output<Node>& data,
         const Output<Node>& gamma,
         double epsilson,
-        const ov::element::Type output_type = ov::element::dynamic);
+        const ov::element::Type output_type = ov::element::dynamic,
+        bool elementwise_affine = true);
```
Comment (e-ddykim) [inline src/common/transformations/include/ov_ops/rms.hpp]: I think we don't need to expose `elementwise_affine` as a parameter since a separate constructor without `gamma` has been added. It would be possible to set `elementwise_affine` to `false` if `gamma` doesn't exist, and to `true` if `gamma` exists.
In addition to this, it would be better to use only one of 'elementwise_affine` or 'has_gamma` as the variable name.

Comment (andrew-k-park) [reply src/common/transformations/include/ov_ops/rms.hpp]: removed `elementwise_affine` parameter from constructors (infer it from gamma's presence) and using consistent naming

Comment (e-ddykim) [reply src/common/transformations/include/ov_ops/rms.hpp]: Thank you!

**Review by e-ddykim — APPROVED** [review]
Looks good to me for the GPU part

**Comment by andrew-k-park** [comment]
@CuriousPanCake Could you review this PR?

File: src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp [hunk 3]
```diff
@@ -87,16 +87,26 @@ RMSFusion::RMSFusion(bool force_tail_convert, bool enable_div_x) {
...
     auto gamma_convert = pattern::optional<v0::Convert>(gamma);
+    auto mul_with_gamma = pattern::wrap_type<v1::Multiply>({gamma_convert, mul_or_div});
 
-    auto mul2 = pattern::wrap_type<v1::Multiply>({gamma_convert, mul_or_div});
+    // Pattern 2: RMS without gamma, but multiplied with dynamic input
+    // RMS(x) * scale where scale is non-constant (e.g., gate, activation, residual)
+    // This allows partial fusion: only fuse up to mul_or_div
+    auto scale = pattern::any_input([](const Output<Node>& value) -> bool {
+        return !ov::is_type<v0::Constant>(value.get_node_shared_ptr());
+    });
```
Comment (CuriousPanCake) [inline src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp]: Can we use `class_other_than()` predicate here?
https://github.com/openvinotoolkit/openvino/blob/0d3c90684d96ecfd0aff4727334d574c08a167db/src/core/include/openvino/pass/pattern/op/pattern.hpp#L40

File: src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp [hunk 4]
```diff
@@ -87,16 +87,26 @@ RMSFusion::RMSFusion(bool force_tail_convert, bool enable_div_x) {
...
+    auto mul_with_scale = pattern::wrap_type<v1::Multiply>({mul_or_div, scale});
 
-    std::shared_ptr<ov::Node> comp = mul2;
+    auto rms_mul = std::make_shared<pattern::op::Or>(OutputVector{mul_with_gamma, mul_with_scale});
+
+    std::shared_ptr<ov::Node> comp = rms_mul;
     if (force_tail_convert) {
```
Comment (CuriousPanCake) [inline src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp:109]: Can we use `optional<Convert>` here?

File: src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp [hunk 5]
```diff
@@ -114,9 +124,15 @@ RMSFusion::RMSFusion(bool force_tail_convert, bool enable_div_x) {
             return false;
         }
 
-        auto gamma_node = pattern_map.at(gamma).get_node_shared_ptr();
-        if (pattern_map.find(gamma_convert) != pattern_map.end()) {
-            gamma_node = pattern_map.at(gamma_convert).get_node_shared_ptr();
+        auto mul_or_div_node = pattern_map.at(mul_or_div).get_node_shared_ptr();
+        bool elementwise_affine = (pattern_map.find(mul_with_gamma) != pattern_map.end());
```
Comment (CuriousPanCake) [inline src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp]: `.count()` is shorter()

File: src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp [hunk 6]
```diff
@@ -114,9 +124,15 @@ RMSFusion::RMSFusion(bool force_tail_convert, bool enable_div_x) {
...
+
+        std::shared_ptr<ov::Node> gamma_node;
+        if (elementwise_affine) {
+            gamma_node = pattern_map.at(gamma).get_node_shared_ptr();
+            if (pattern_map.find(gamma_convert) != pattern_map.end()) {
```
Comment (CuriousPanCake) [inline src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp]: `.count()` is shorter

File: src/common/transformations/tests/common_optimizations/rms_norm_decomposition_test.cpp [hunk 7]
```diff
@@ -320,3 +320,67 @@ TEST_F(TransformationTestsF, RMSNormFusionTest9) {
     comparator.enable(FunctionsComparator::CmpValues::CONST_VALUES);
     comparator.enable(FunctionsComparator::CmpValues::ATTRIBUTES);
 }
+
+TEST_F(TransformationTestsF, RMSNormFusionTest10) {
```
Comment (CuriousPanCake) [inline src/common/transformations/tests/common_optimizations/rms_norm_decomposition_test.cpp:326]: Try using ov::op::vX rather than opsetX

**Review by CuriousPanCake — CHANGES_REQUESTED** [review]
Please, review the comments for the Transformations part

File: src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp (hunk 3 above)
Comment (andrew-k-park) [reply src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp]: done

File: src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp (hunk 5 above)
Comment (andrew-k-park) [reply src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp]: done

File: src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp (hunk 6 above)
Comment (andrew-k-park) [reply src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp]: done

File: src/common/transformations/tests/common_optimizations/rms_norm_decomposition_test.cpp (hunk 7 above)
Comment (andrew-k-park) [reply src/common/transformations/tests/common_optimizations/rms_norm_decomposition_test.cpp:326]: done

File: src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp (hunk 4 above)
Comment (andrew-k-park) [reply src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp:109]: The current `if (force_tail_convert)` implementation provides mutually exclusive behavior:

- When `force_tail_convert=true`: Convert must exist at the end (required for compression to f16)
- When `force_tail_convert=false`: Convert must not exist (keep f32 precision)

If we use `pattern::optional<Convert>`, it would match both cases regardless of the flag, breaking the intended behavior:

The current implementation is correct and should remain as-is.

**Comment by pereanub** [comment]
@dmatveev, this PR is breaking some NPUW func test. Please have a look.

**Comment by andrew-k-park** [comment]
> @dmatveev, this PR is breaking some NPUW func test. Please have a look.
In my opinion, this NPU test appears to be related to the CPU plugin’s transformation pipeline. In the CPU plugin, I have made a change so that the relevant pattern is no longer matched in RMSFusion.

**Comment by andrew-k-park** [comment]
@CuriousPanCake Could you please review the updated code that apply your comments again?

File: src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp (hunk 4 above)
Comment (CuriousPanCake) [reply src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp:109]: I see, you don't want to match and enter the callback in some cases. Thanks

File: src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp [hunk 8]
```diff
@@ -129,11 +147,30 @@ RMSFusion::RMSFusion(bool force_tail_convert, bool enable_div_x) {
...
-        ov::copy_runtime_info(m.get_matched_nodes(), rms);
-        ov::replace_node(m.get_match_root(), rms);
+        auto output_type = elementwise_affine ? m.get_match_root()->get_output_element_type(0)
+                                              : mul_or_div_node->get_output_element_type(0);
+        std::shared_ptr<ov::op::internal::RMS> rms;
+        if (elementwise_affine) {
+            rms = std::make_shared<ov::op::internal::RMS>(x_output, gamma_node, eps_value, output_type);
+        } else {
+            rms = std::make_shared<ov::op::internal::RMS>(x_output, eps_value, output_type);
+        }
```
Comment (CuriousPanCake) [inline src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp]: Minor: consider writing usinga ternary operator here. Might be a little shorter.

**Comment by CuriousPanCake** [comment]
LGTM from the transformations side

File: src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp (hunk 8 above)
Comment (andrew-k-park) [reply src/common/transformations/src/transformations/common_optimizations/rms_fusion.cpp]: applied
